diff.go

```diff
@@ -118,6 +119,9 @@ type DiffFile struct {
 
 	oldName string
 
+	Mode    EntryMode
+	OldMode EntryMode
```

Comment:
These two are public fields, please move up to near after the `Index` and `OldIndex`, also missing docstrings.

Reply:
i've change those into r/o functions and added docstrings